### PR TITLE
Update order of the gallery image info caption or description

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator : https://phabricator.wikimedia.org/T262074

Problem
===

The gallery image has caption, description and also caption from the media list api.

Solution
===

The api doesn't return the caption, therefore first choose the media info description then media list caption.

### CODE REVIEW

**Review by stephanebisson — APPROVED** [review]

**Review by medied — APPROVED** [review]

File: src/gallery/slider.js [hunk 1]
```diff
@@ -53,10 +53,14 @@ const clientWidth = window.innerWidth,
 
 	renderImageInfo = ( mediaInfo, image, lang ) => {
 		const getImageDescription = () => {
-				if ( image.caption ) {
-					return image.caption
-				} else if ( mediaInfo.description ) {
+				// description list order
```
Comment (jpita) [inline src/gallery/slider.js:56]: should we delete the comments?

Comment (hueitan) [reply src/gallery/slider.js:56]: i think it's better to keep as the memory

**Comment by jpita** [comment]
[This image](https://en.wikipedia.org/wiki/Ivory#/media/File:The_Bull_Leaper_Knossos_1500BC.jpg) shows with empty author and description on wikipedia preview 
image.

[In commons](https://commons.wikimedia.org/wiki/File:The_Bull_Leaper_Knossos_1500BC.jpg) there's a description and an author.


Is this expected or a bug?

**Comment by hueitan** [comment]
> [This image](https://en.wikipedia.org/wiki/Ivory#/media/File:The_Bull_Leaper_Knossos_1500BC.jpg) from the word "ivory" shows with empty author and description on wikipedia preview
> image.
> 
> [In commons](https://commons.wikimedia.org/wiki/File:The_Bull_Leaper_Knossos_1500BC.jpg) there's a description and an author.
> 
> Is this expected or a bug?

Checking on the API, it returns no description and licenese, therefore expected, same behaviour in iOS app.

**Pull request merged**